### PR TITLE
Add V1-V2 token swap functionality

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

This PR adds the V1-V2 token swap functionality

⚠️ This feature is behind the `v1TokenSwap` feature flag, disabled by default.

Specifically:
- Add a "Your V1 tokens" to the **Tokens** section of the V2 project page
- Add a modal to facilitate the swap.

Closes https://github.com/jbx-protocol/juice-interface/issues/1335
(related to https://github.com/jbx-protocol/juice-interface/issues/1056)

### Tokens section

| default | tooltip |
| --- | --- |
| Screen Shot 2022-07-14 at 2 57 28 PM | Screen Shot 2022-07-14 at 2 57 32 PM |

### Swap modal

| Step 1. Give permission to swap | Step 2. Swap form |
| --- | --- |
| Screen Shot 2022-07-14 at 3 00 14 PM | Screen Shot 2022-07-14 at 2 57 39 PM |

### Token swap

http://localhost:3000/#/v2/p/4284

1.  Enable the feature flag

    ```js
    localStorage.setItem('v1TokenSwap_rinkeby', 'true')
    ```
4. Pay this V1 project to receive v1 tokens: http://localhost:3000/#/p/testballots
5. Navigate to http://localhost:3000/#/v2/p/4284. You should see the new "V1 token balance" in the **Tokens** section.
6. Click **Swap for V2 tokens**
7. Click **Give permission**
8. Swap some tokens.
9. Verify that the new V1 and V2 token balances are correct.

### End to end (optional)

1.  Enable the feature flag

    ```js
    localStorage.setItem('v1TokenSwap_rinkeby', 'true')
    ```
2. Create a V1 and a V2 project.
3. On the V2 project, configure it for V1 token migration
   i ) Go to **Tools** > **Owner tools** > **V1 token migration**
   ii ) Select **Set up token migration**
   iii ) Complete the modal.
4. Pay the V1 project to receive V1 tokens.
5. Visit the V2 project. You should see the new "V1 token balance" in the **Tokens** section.
6. Click **Swap for V2 tokens**
7. Click **Give permission**
8. Swap some tokens.
9. Verify that the new V1 and V2 token balances are correct.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by aeolianeth** [comment]
@johnnyd-eth 👋 , could you please review this when you get a chance? You'll have the most context for this 🙏

File: src/components/TextButton.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Button, ButtonProps, Space } from 'antd'
+import { RightCircleOutlined } from '@ant-design/icons'
+import { useContext } from 'react'
+import { ThemeContext } from 'contexts/themeContext'
+
+export function TextButton({ style, ...props }: ButtonProps) {
```
Comment (aeolianeth) [inline src/components/TextButton.tsx:6]: Smol change.

We use this TextButton for the `ASSETS` button, and now the `HOLDERS` button

File: src/components/modals/ParticipantsModal.tsx [hunk 2]
```diff
@@ -271,52 +272,53 @@ export default function ParticipantsModal({
             holders
           </Trans>
         </h4>
+        <Space direction="vertical">
```
Comment (aeolianeth) [inline src/components/modals/ParticipantsModal.tsx:275]: random change as I was passing by. Uses `Callout` and wraps the thing in `Space`

File: src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/MigrateV1ProjectTokensModal/GrantTransferPermissionCallout.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { Button } from 'antd'
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/MigrateV1ProjectTokensModal/GrantTransferPermissionCallout.tsx:1]: This component renders this UI

Screen Shot 2022-07-14 at 3 00 14 PM

File: src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/MigrateV1ProjectTokensModal/MigrateV1ProjectTokensForm.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { Trans } from '@lingui/macro'
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/MigrateV1ProjectTokensModal/MigrateV1ProjectTokensForm.tsx:1]: This form is:

**Tokens to swap**

Screen Shot 2022-07-14 at 2 57 39 PM

File: src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/V1ProjectTokensSection.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import { Trans } from '@lingui/macro'
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ManageTokensSection/V1ProjectTokensSection/V1ProjectTokensSection.tsx:1]: This UI is the **Your V1 balance**

Screen Shot 2022-07-14 at 2 57 28 PM

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx [hunk 6]
```diff
@@ -110,7 +111,7 @@ export function V2ConfirmPayModal({
   )
   let nftRewardTier: NftRewardTier | null = null
 
-  if (nftRewardTiers && featureFlagEnabled('nftRewards')) {
+  if (nftRewardTiers && featureFlagEnabled(FEATURE_FLAGS.NFT_REWARDS)) {
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx:114]: Introduce single source of truth for feature flag names

File: src/constants/featureFlags.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export const FEATURE_FLAGS = {
```
Comment (aeolianeth) [inline src/constants/featureFlags.ts:1]: new

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx (hunk 6 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx:114]: Sweet

File: src/constants/featureFlags.ts (hunk 7 above)
Comment (johnnyd-eth) [reply src/constants/featureFlags.ts:1]: gr8

**Comment by johnnyd-eth** [comment]
Code all looks great. Although I'm getting an error on step 5:

https://user-images.githubusercontent.com/96150256/179127797-97df7225-039b-4a3a-b67b-f81fbb527a74.mp4

**Comment by johnnyd-eth** [comment]
Screen Shot 2022-07-15 at 2 52 58 pm

`span` missing

**Review by johnnyd-eth — APPROVED** [review]
Legend LGTM

**Pull request merged**